tests/playwright/src/specs/podman-machine-rootless.spec.ts

```diff
@@ -0,0 +1,75 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import * as os from 'node:os';
+
+import type { Page } from '@playwright/test';
+import { expect as playExpect } from '@playwright/test';
+import { afterAll, beforeAll, beforeEach, describe, test } from 'vitest';
+
+import { CreateMachinePage } from '../model/pages/create-machine-page';
+import { ResourcesPage } from '../model/pages/resources-page';
+import { ResourcesPodmanConnections } from '../model/pages/resources-podman-connections-page';
+import { WelcomePage } from '../model/pages/welcome-page';
+import { NavigationBar } from '../model/workbench/navigation';
+import { PodmanDesktopRunner } from '../runner/podman-desktop-runner';
+import type { RunnerTestContext } from '../testContext/runner-test-context';
+import { deletePodmanMachine } from '../utility/operations';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let navBar: NavigationBar;
+const PODMAN_MACHINE_NAME: string = 'podman-machine-rootless';
+const MACHINE_VISIBLE_NAME: string = 'Podman Machine rootless';
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('podman-rootless-machine-e2e');
+
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  navBar = new NavigationBar(page);
+});
+
+afterAll(async () => {
+  await deletePodmanMachine(page, PODMAN_MACHINE_NAME);
+  await pdRunner.close();
+});
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+describe.skipIf(os.platform() === 'linux')('Rootless Podman machine Verification', async () => {
+  test('Create a rootless machine', async () => {
+    await navBar.openSettings();
+    const resourcesPage = new ResourcesPage(page);
+
+    const createMachineButton = resourcesPage.podmanResources.getByRole('button', {
```

Comment:
If there will be more that one Resource provider it will fail on ambiguity. 
update the logic so it finds the podman provider. Look under `model/pages/` and search for `resource-page.ts` and `resources-podman-connections-page.ts`. There will be new part besides `this.providerConnections = this.podmanResources.getByRole('region', { name: 'Provider Connections' });` : `Provider Setup`.

Reply:
nah, it should not have, since you can have more than one machine. Ok then!